chore: Add common steps to utilities

dsimansk opened the pull request:
Resolves #207 

This PR:
- creates new sub-package `bdd` in utilities to accommodate commons steps 
- changes the BDD binding to directive style (seems more natural as this action is not performed by "user" directly)
- adds `verifyBuild method`
  - `@Then("^verify s2i build of integration \"([^\"]*)\" was finished in duration (\\d+) min$")`

tplevko commented on utilities/src/main/java/io/syndesis/qe/templates/SyndesisTemplate.java:
```diff
@@ -22,8 +22,9 @@
-	private static final String SUPPORT_SA_URL = "https://raw.githubusercontent.com/syndesisio/syndesis-openshift-templates/master/support/serviceaccount-as-oauthclient-restricted.yml";
-	private static final String TEMPLATE_URL = "https://raw.githubusercontent.com/syndesisio/syndesis-openshift-templates/master/syndesis-restricted.yml";
+	//TODO: move this to TestConfiguration
+	private static final String SUPPORT_SA_URL = "https://raw.githubusercontent.com/syndesisio/syndesis/master/deploy/support/serviceaccount-as-oauthclient-restricted.yml";
```
fix the URL

tplevko commented on utilities/src/main/java/io/syndesis/qe/templates/SyndesisTemplate.java:
```diff
@@ -22,8 +22,9 @@
-	private static final String TEMPLATE_URL = "https://raw.githubusercontent.com/syndesisio/syndesis-openshift-templates/master/syndesis-restricted.yml";
+	//TODO: move this to TestConfiguration
+	private static final String SUPPORT_SA_URL = "https://raw.githubusercontent.com/syndesisio/syndesis/master/deploy/support/serviceaccount-as-oauthclient-restricted.yml";
+	private static final String TEMPLATE_URL = "https://raw.githubusercontent.com/syndesisio/syndesis/master/deploy/syndesis-restricted.yml";
```
fix URL

tplevko commented on ui-tests/src/test/java/io/syndesis/qe/CucumberTest.java:
```diff
@@ -1,32 +1,25 @@
 		features = "classpath:features",
-		format = {"pretty"}
+		format = {"pretty", "html:target/cucumber-report.html", "junit:target/cucumber-junit.html"}
 )
```
I needed some glue here: glue = {"io.syndesis.qe.bdd"})

dsimansk commented:
@tplevko I've fixed those changed URLs and also made them more configurable. The second issue with missing glue is also fixed. As we've discussed on IRC might be worth revisit the package model to make the common steps auto-discoverable maybe?

tplevko commented on utilities/src/main/java/io/syndesis/qe/TestConfiguration.java:
```diff
@@ -29,6 +29,9 @@
 	public static final String SYNDESIS_VERSIONS_FILE = "syndesis.config.versions.file";
 
+	public static final String SYNDESIS_TEMPLATE_URL = "syndesis.config.template.url";
+	public static final String SYNDESIS_TEMPLATE_SA = "syndesis.config.template.sa.";;
```
cosmetic issue: ";;" and "syndesis.config.template.sa." - remove dot from the end

dsimansk commented:
@tplevko all done hopefully.
